key.go

```diff
@@ -162,15 +162,69 @@ func New(modulePath, tokenLabel, pin, privateKeyLabel string) (*Key, error) {
 		module:     module,
 		tokenLabel: tokenLabel,
 		pin:        pin,
+		publicKey:  publicKey,
 	}
-	err = ps.setup(privateKeyLabel)
+
+	err = ps.setup()
 	if err != nil {
 		return nil, fmt.Errorf("pkcs11key: %s", err)
 	}
 	return ps, nil
 }
 
-func (ps *Key) setup(privateKeyLabel string) error {
+// findObject finds an object in the PKCS#11 token according to a template. It
+// returns error if there are more than one result, or if there was an error
```

Comment:
typo: "returns error if there are more than one result" -> "returns error if there is more than one result"